### ISSUE TITLE
json: cannot unmarshal object error 

### ISSUE THREAD

**Issue by steven004** [filecoin-project/lotus#601]
# Version
commit: 7150ee811d14aa4f7e79aa85d0d7420cb7bc8149

# Description
The following error happens when getting randomness for sealTicket: 
```
2019-11-14T19:01:15.885+0800	INFO	storageminer	storage/sealing.go:177	Sector 4 updated state to PreCommitting
2019-11-14T19:01:15.885+0800	ERROR	storageminer	storage/sealing.go:188	sector 4 failed: getting randomness for SealTicket failed: RPC error (-32700): unmarshaling params for '%!s(func(*api.FullNodeStruct, context.Context, types.TipSetKey, []*types.Ticket, int) ([]uint8, error)=0x4655af0)': json: cannot unmarshal object into Go value of type []cid.Cid
github.com/filecoin-project/lotus/storage.(*Miner).onSectorUpdated
	/Users/steven/go/src/github.com/filecoin-project/lotus/storage/sealing.go:188
github.com/filecoin-project/lotus/storage.(*Miner).sectorStateLoop.func1
	/Users/steven/go/src/github.com/filecoin-project/lotus/storage/sealing.go:139
2019-11-14T19:01:15.885+0800	ERROR	storageminer	storage/sealing.go:218	sector 4 error: getting randomness for SealTicket failed:
    github.com/filecoin-project/lotus/node/modules.SealTicketGen.func1
        /Users/steven/go/src/github.com/filecoin-project/lotus/node/modules/storageminer.go:226
  - RPC error (-32700): unmarshaling params for '%!s(func(*api.FullNodeStruct, context.Context, types.TipSetKey, []*types.Ticket, int) ([]uint8, error)=0x4655af0)': json: cannot unmarshal object into Go value of type []cid.Cid
github.com/filecoin-project/lotus/storage.(*Miner).failSector
	/Users/steven/go/src/github.com/filecoin-project/lotus/storage/sealing.go:218
github.com/filecoin-project/lotus/storage.(*Miner).onSectorUpdated
	/Users/steven/go/src/github.com/filecoin-project/lotus/storage/sealing.go:189
github.com/filecoin-project/lotus/storage.(*Miner).sectorStateLoop.func1
	/Users/steven/go/src/github.com/filecoin-project/lotus/storage/sealing.go:139
```

**Comment by magik6k** [comment]
Can you see if changing https://github.com/filecoin-project/lotus/blob/4f9947a27b974512477342ce092381c40a5e8ac9/chain/types/tipset_key.go#L81 to

```go
return json.Marshal(&k.Cids())
```

helps?

**Comment by anorth** [comment]
I think it's fixed in #610

**Comment by Kubuxu** [comment]
Resolved by #610
